test/precompile.jl

```diff
@@ -875,3 +875,19 @@ precompile_test_harness("Renamed Imports") do load_path
     Base.compilecache(Base.PkgId("RenameImports"))
     @test (@eval (using RenameImports; RenameImports.test())) isa Module
 end
+
+@testset "issue 38149" begin
+    M = Module()
+    @eval M begin
+        @nospecialize
+        f(x, y) = x + y
+        f(x::Int, y) = 2x + y
+    end
+    precompile(M.f, (Int, Any))
+    precompile(M.f, (AbstractFloat, Any))
```

Comment:
What do you think about testing the `precompile` results directly ?

```suggestion
    @test precompile(M.f, (Int, Any))
    @test precompile(M.f, (AbstractFloat, Any))
```

Reply:
`precompile` was changed to always return true.

Reply:
ahh, yeah I forgot that.